### PR TITLE
CI: bump mypy 0.730

### PR DESCRIPTION
**Pull request by simonjayhawkins**
xref https://github.com/pandas-dev/pandas/pull/29188#issuecomment-554524916

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -145,10 +145,13 @@ ignore_errors=True
 [mypy-pandas.tests.extension.json.test_json]
 ignore_errors=True
 
+[mypy-pandas.tests.indexes.datetimes.test_tools]
+ignore_errors=True
+
 [mypy-pandas.tests.indexes.test_base]
 ignore_errors=True
 
-[mypy-pandas.tests.indexing.test_loc]
```
Comment (simonjayhawkins) [inline setup.cfg]: this is a false positive in 0.720 so can be removed

File: setup.cfg [hunk 2]
```diff
@@ -145,10 +145,13 @@ ignore_errors=True
 [mypy-pandas.tests.extension.json.test_json]
 ignore_errors=True
 
+[mypy-pandas.tests.indexes.datetimes.test_tools]
```
Comment (simonjayhawkins) [inline setup.cfg]: xref #28932, add `#type: ignore` as an alternative to ignoring all errors in module

File: setup.cfg [hunk 3]
```diff
@@ -145,10 +145,13 @@ ignore_errors=True
...
+ignore_errors=True
+
 [mypy-pandas.tests.indexes.test_base]
 ignore_errors=True
 
-[mypy-pandas.tests.indexing.test_loc]
+[mypy-pandas.tests.scalar.period.test_period]
```
Comment (simonjayhawkins) [inline setup.cfg]: this is now giving `error: Module has no attribute "__version__"` following #29212

again could add `#type: ignore`, see https://github.com/pandas-dev/pandas/pull/28339/commits/cfc2ef78d24aab0fbc4efd838e702ef8cbecb19f

File: setup.cfg (hunk 2 above)
Comment (simonjayhawkins) [reply setup.cfg]: a 'cleaner' fix in https://github.com/pandas-dev/pandas/pull/28339/commits/98491632b70ae319afbb14c04e5740f06bedd89e, that could be simply reverted when `--warn-unused-ignores` reports issue fixed.

**Review by WillAyd — APPROVED** [review]
lgtm whenever travis is green

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**